src/app/search/search.component.ts

```diff
@@ -20,152 +22,105 @@ export class SearchComponent implements OnInit {
+      this.auth.followItems('prx:series', {filters: 'v4'}).subscribe((series) => {
+        for (let s of series) {
+          this.allSeriesIds.push(s.id);
+          this.allSeries[s.id] = new SeriesModel(this.auth, s, false);
         }
+        this.subscribeRouteParams();
       });
     });
   }
 
-  initStorySearch() {
-    this.allSeriesIds = [-1];
-    this.allSeries = {};
-    this.auth.followItems('prx:series', {filters: 'v4'}).subscribe((series) => {
-      for (let s of series) {
-        this.allSeriesIds.push(s.id);
-        this.allSeries[s.id] = new SeriesModel(this.auth, s, false);
-      }
-
-      this.searchTextParam = '';
-      this.searchSeriesParam = null;
-      this.currentPage = 1;
-      if (this.searchSeriesIdParam) {
-        this.searchStoriesBySeries(this.searchSeriesIdParam);
-      } else {
-        this.searchStories(1);
+  subscribeRouteParams() {
+    this.route.params.forEach((params) => {
+      this.selectedTab = params['tab'] || SearchComponent.TAB_STORIES;
+      this.currentPage = params['page'] ? +params['page'] : 1;
+      if (this.selectedTab === SearchComponent.TAB_STORIES) {
+        this.searchStoryParams.fromRouteParams(params);
+        this.searchStories();
+      } else if (this.selectedTab === SearchComponent.TAB_SERIES) {
+        this.searchSeriesParams.fromRouteParams(params);
+        this.searchSeries();
       }
     });
   }
 
-  initSeriesSearch() {
-    this.searchSeries(1);
-  }
-
-  searchByText(text: string) {
-    this.searchTextParam = text;
-    this.search(1);
-  }
-
-  searchByOrder(order: any) {
-    this.searchOrderByParam = order.orderBy;
-    this.searchOrderDescParam = order.desc;
-    this.search(1);
-  }
-
-  searchStoriesBySeries(searchSeriesId: number) {
-    this.searchSeriesIdParam = searchSeriesId;
-    this.searchSeriesParam = this.allSeries[this.searchSeriesIdParam];
-    this.searchStories(1);
-  }
-
-  searchSeriesByGenre(genre: any) {
-    this.searchGenreParam = genre.genre;
-    this.searchSubGenreParam = genre.subgenre;
-    this.searchSeries(1);
+  searchWithParams(searchParams: any) {
+    let routeParams = {tab: this.selectedTab};
+    Object.keys(searchParams).forEach(key => {
+      if (searchParams[key] !== undefined) {
+        routeParams[key] = searchParams[key];
+      }
+    });
+    this.router.navigate(['search', routeParams]);
   }
 
-  search(page: number, per: number = undefined) {
-    if (this.isOnSeriesTab()) {
-      this.searchSeries(page, per);
-    } else if (this.isOnStoriesTab) {
-      this.searchStories(page, per);
+  navigateToPagePer(page: number, perPage: number = undefined) {
+    if (this.selectedTab === SearchComponent.TAB_STORIES) {
+      this.searchWithParams(Object.assign({}, this.searchStoryParams, {tab: this.selectedTab, page, perPage}));
+    } else if (this.selectedTab === SearchComponent.TAB_SERIES) {
+      this.searchWithParams(Object.assign({}, this.searchSeriesParams, {tab: this.selectedTab, page, perPage}));
```

Comment:
I was worried that I might need a polyfill for Object (we did on a Babel project) but typescript takes care of it. I miss having the spread operator for objects though, which I have a feeling will make me sad when doing Redux.